### ISSUE TITLE
Optimal Learning Rate when the number of train step doubles?

### ISSUE THREAD

**Issue by minstar** [pytorch/fairseq#1225]
Hi, I implemented the simple adversarial training scheme with fairseq library. After one train step from ```trainer.train_step(samples)``` from train.py, with the use of gradient vectors from the model, I  generated adversarial examples and then trained my model with calling ```trainer.train_step(adversarial_samples)``` again. This question might be silly, but since I doubled the number of train_steps for the same dataset, I think I need to finetune the learning rate by increasing it. (maybe by doubling the learning rate?). Is it right? Or any problems I made with my implementation?

Thank you!

**Comment by alexeib** [comment]
in general the learning rate is a function of the model size, the task, and the batch size, not the length of training. you can try playing around with various learning rate schedules (e.g. cosine, polynomial_decay, etc) to make sure that the learning rate decays uniformly over the course of training, but aside from that tuning learning rate is usually done experimentally
